Coverage

dmcnulla opened the pull request:
Added code to check test coverage. 

Malo, you are welcome to review the changes, but not required. I will push the changes to master in 2 days if I do not hear from you. Thanks.

Commit:
Add code coverage to the system.

Commit:
add ci

dmcnulla committed:
Update env.rb

Add requires for coverage.

dmcnulla committed:
Update .coveralls.yml

dmcnulla committed:
Update .coveralls.yml

Commit:
Update Rake for coverage

Commit:
Merged trunk to coverage branch.

dmcnulla committed:
Update README.md

Added coverage badge.

dmcnulla committed:
Update README.md

majioa commented on Gemfile:
```diff
@@ -6,6 +6,8 @@ gem 'ruby-fifo', github: 'shurizzle/ruby-fifo'
 group :development do
   gem 'rspec'
+  gem 'coveralls'
+  gem 'simplecov'  
```
Its better to add development dependencies that do not require a dynamic references to guthub, etc, to gemspec.

dmcnulla replied:
I'm not sure what the means. Can you give an example? Thanks.

majioa replied:
@dmcnulla I meant that to do as here: https://github.com/dmcnulla/ntp/blob/master/ntp-mock-server.gemspec#L30

dmcnulla replied:
Thanks. I had forgotten that.

majioa replied:
Also since they were added to gemspec the can be removed from Gemfile (except lines belongs to gem that referenced to github).btw, why do you need the rspec and rubocop?

dmcnulla replied:
I did not realize that Gemfile and .gemspec served the same purpose. I will remove them from Gemfile. 

I use rubocop for stylechecker/linter. rspec handles the expectations (asserts) I use in cucumber. I will move those to gemspec as well.

majioa replied:
@dmcnulla

> I did not realize that Gemfile and .gemspec served the same purpose. I will remove them from Gemfile. 

Gemspec server to declare valid dependencies for a gem specifically, while Gemfile is mostly just for a currentinstallation locally. So rubygems will use only dependencies from gemspec.

> I use rubocop for stylechecker/linter. rspec handles the expectations (asserts) I use in cucumber. I will move those to gemspec as well. 

Yes but I haven't seen any rspec file in the repo.

dmcnulla replied:
https://github.com/dmcnulla/ntp/blob/coverage/features/step_definitions/server_steps.rb 
Lines 17 & 28 use expect(). This is an rspec method.
...
OK. I replaced rspec with rspec-expecations. The test failed because it required rspec/mock. So I reverted that change. I think rspec is needed.

majioa commented on README.md:
```diff
@@ -1,6 +1,7 @@
 # NTP Mock Server
 
-[![Build Status](https://travis-ci.org/dmcnulla/ntp.svg?branch=master)](https://travis-ci.org/dmcnulla/ntp)
+[![Build Status](https://travis-ci.org/dmcnulla/ntp.svg?branch=coverage)](https://travis-ci.org/dmcnulla/ntp)
```
why did you specified coverage branch?

dmcnulla replied:
That just says to provide the report for this branch in this readme. In the master, it will be master.

majioa replied:
yes, I see, but if you merges coverage into the master branch, the changes will belong to coverage anyway.

dmcnulla replied:
I switched it to master.

Commit:
Added versions to match the gemspec.

Commit:
Updates per code review comments.

Commit:
Updated Rakefile to include coveralls references.

dmcnulla commented:
I updated requested changes. I also noticed the gemspec indicates this server supports >= 1.9.3 so I added those to the travis-ci spec. I will merge tomorrow unless you have additional comments.

dmcnulla committed:
Moved gem requirements to gemspec from gemfile

Commit:
Replaced rspec gem reference with rspec-expectations

Commit:
Revert "Replaced rspec gem reference with rspec-expectations"

This reverts commit 6d2c5b6b19e0600d9388e39211909dd87763b6a3.

majioa commented on Gemfile:
```diff
@@ -3,11 +3,3 @@ source 'https://rubygems.org'
 gem 'ruby-fifo', github: 'shurizzle/ruby-fifo'
-
-group :development do
-  gem 'rspec'
```
@dmcnulla  Have you use the rspec and rubocop just locally for the project?

dmcnulla replied:
I use rubocop locally, like you use pry. Cucumber uses the expect method of
rspec. Tests will not run in travis-ci if rspec is not included.

Thanks,

Dave McNulla

On Fri, Apr 8, 2016 at 10:42 PM, Павел Скрылёв (Malo Skrylevo) <
notifications@github.com> wrote:

> In Gemfile https://github.com/dmcnulla/ntp/pull/1#discussion_r59110829:
> 
> > @@ -3,11 +3,3 @@ source 'https://rubygems.org'
> >  gemspec
> > 
> > ##  gem 'ruby-fifo', github: 'shurizzle/ruby-fifo'
> > 
> > -group :development do
> > -  gem 'rspec'
> 
> @dmcnulla https://github.com/dmcnulla Have you use the rspec and
> rubocop just locally for the project?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/dmcnulla/ntp/pull/1/files/24a259dc48a3962964a53e46b3778e6441c8ae2c..1ec80f8d6a1bf2887d99a8bd6f079fbd1167a85d#r59110829

majioa replied:
well then, I believe, pry, rubocop, and rspec can be moved back to Gemfile, in order to not waste gemspec =)

dmcnulla replied:
The rspec has to stay with the cucumber or the tests will not run:
https://travis-ci.org/dmcnulla/ntp/builds/121730524

I can move the other to Gemfile.

Thanks,

Dave McNulla

On Fri, Apr 8, 2016 at 10:52 PM, Павел Скрылёв (Malo Skrylevo) <
notifications@github.com> wrote:

> In Gemfile https://github.com/dmcnulla/ntp/pull/1#discussion_r59110909:
> 
> > @@ -3,11 +3,3 @@ source 'https://rubygems.org'
> >  gemspec
> > 
> > ##  gem 'ruby-fifo', github: 'shurizzle/ruby-fifo'
> > 
> > -group :development do
> > -  gem 'rspec'
> 
> well the I believe pry, rubocop, and rspec can be moved back to Gemfile,
> in order to not waste gemspec =)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/dmcnulla/ntp/pull/1/files/24a259dc48a3962964a53e46b3778e6441c8ae2c..1ec80f8d6a1bf2887d99a8bd6f079fbd1167a85d#r59110909

dmcnulla replied:
Sorry, I accidentally merged. It's late at night here and I'm tired. I
added the last change to the Gemfile & .gemspec on master. Not how I wanted
to do it but it's working now. I don't mind you making other changes as
long as it's OK on CI.

BTW, I found CI failed once or twice due to a timing issue. You can see it
in the logs when that happens. There is a rerun button (I'm not sure if you
can re-run it, but you can let me know if it is an issue). All other
failures on CI that I saw were reporting problems with gems or ruby
versions.

Thanks,

Dave McNulla

On Fri, Apr 8, 2016 at 10:58 PM, Dave McNulla mcnulla@gmail.com wrote:

> The rspec has to stay with the cucumber or the tests will not run:
> https://travis-ci.org/dmcnulla/ntp/builds/121730524
> 
> I can move the other to Gemfile.
> 
> Thanks,
> 
> Dave McNulla
> 
> On Fri, Apr 8, 2016 at 10:52 PM, Павел Скрылёв (Malo Skrylevo) <
> notifications@github.com> wrote:
> 
> > In Gemfile https://github.com/dmcnulla/ntp/pull/1#discussion_r59110909:
> > 
> > > @@ -3,11 +3,3 @@ source 'https://rubygems.org'
> > >  gemspec
> > > 
> > > ##  gem 'ruby-fifo', github: 'shurizzle/ruby-fifo'
> > > 
> > > -group :development do
> > > -  gem 'rspec'
> > 
> > well the I believe pry, rubocop, and rspec can be moved back to Gemfile,
> > in order to not waste gemspec =)
> > 
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly or view it on GitHub
> > https://github.com/dmcnulla/ntp/pull/1/files/24a259dc48a3962964a53e46b3778e6441c8ae2c..1ec80f8d6a1bf2887d99a8bd6f079fbd1167a85d#r59110909

majioa replied:
> BTW, I found CI failed once or twice due to a timing issue. 

I know the problem, the code which will wait for an answer when a new time base is set, should be added. But it is as far as I can
